src/olympia/users/tests/test_tasks.py

```diff
@@ -172,3 +179,136 @@ def test_resize_photo_poorly():
+            f'{settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc',
+            callback=timeout_callback,
+        )
+
+        task = sync_blocked_emails.s()
+
+        with pytest.raises(Retry):
+            task.apply(throw=True)
+
+    def test_empty_csv(self):
+        csv = list_to_csv([])
+
+        responses.add(
+            responses.GET,
+            f'{settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc',
+            body=csv,
+            status=200,
+        )
+
+        sync_blocked_emails()
+
+        assert EmailBlock.objects.count() == 0
+
+    def test_existing_email(self):
+        user = user_factory()
+
+        csv = list_to_csv([user.email])
+
+        responses.add(
+            responses.GET,
+            f'{settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc',
+            body=csv,
+            status=200,
+        )
+
+        EmailBlock.objects.create(email=user.email)
+
+        sync_blocked_emails()
+
+        email_block = EmailBlock.objects.get(email=user.email)
+
+        assert email_block.email == user.email
+        assert EmailBlock.objects.count() == 1
+
+    def test_unique_email(self):
+        user = user_factory()
+
+        csv = list_to_csv([user.email])
+
+        responses.add(
+            responses.GET,
+            f'{settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc',
+            body=csv,
+            status=200,
+        )
+
+        sync_blocked_emails()
+
+        email_block = EmailBlock.objects.get(email=user.email)
+
+        assert email_block.email == user.email
+        assert EmailBlock.objects.count() == 1
+
+    @mock.patch.object(uuid, 'uuid4', return_value=static_id)
+    def test_delete_file_after_processing_records(self, mock_uuid):
+        csv = list_to_csv([])
+
+        responses.add(
+            responses.GET,
+            f'{settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc',
+            body=csv,
+            status=200,
+        )
+
+        sync_blocked_emails()
+
+        file_path = f'tmp/suppressions-{mock_uuid.return_value}.csv'
+
+        assert not default_storage.exists(file_path)
+
+        assert EmailBlock.objects.count() == 0
+
+
+"""
+Test scenarios:
+2. recieve bad response from api
+3. cannot reach API, timeout
+5. CSV is empty
+6. CSV contains existing email
+7. CSV contains unique email
+10. Deletes file if error occurs while processing recores.
+
+
+1. environment variables are undefined.
+4. File is deleted before able to read it.
+8. CSV contains same email twice
+9. CSV contains many records.
+"""
```

Comment:
I'm not sure what this comment is - is it explaining what the tests above do?  Or an extended #TODO?

Reply:
It's an extended todo. I added a [comment](https://github.com/mozilla/addons-server/pull/21556#discussion_r1415568442) explaining why it is there for now. I'm still working on making the tests for this PR right, and the bottom chunk are things that could be tested, but maybe shouldn't. Wanted to get some feedback as to how to handle them.